### ISSUE TITLE
Spreading an array inside an object where an array is expected doesn't cause a type error

### ISSUE THREAD

**Issue by lucaslcode** [microsoft/TypeScript#49667]
# Bug Report

### 🔎 Search Terms

destructure, spread, object, array

### 🕗 Version & Regression Information
Using 4.5.5 but also in nightly playground

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about spread operator

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.8.0-dev.20220624#code/C4TwDgpgBAsiAq5oF4oG9gQM7AFxRwCcBLAOwHMBfAWACg6BjAe1JygBtid85FIBtALpRU-DNjxQARAEMplADTpM3aQCN5ggNx1GLNgA98AJQjNCAEwA8RMuSW8kQgHwj0dKNIAWEduyZS+GgenlAAdBGcOCE0tLF6rEzsEGH+5AAUBgCUOrQG-FI+fgGCYQC2MmDp6b5ZIq61WkA)

### 💻 Code

```ts
type MyType = {test: string}

const list: MyType[] = [{test: "a"}, {test: "b"}];

const x: Record<string, MyType[]> = {
  // object shouldn't be allowed?
  "hello": {
    ...list
  }
}

console.log(x);
x["hello"].map((el) => el);
// map doesn't exist on object
```

### 🙁 Actual behavior

Spreading an array inside an object where an array is expected works, and the type information is then incorrect.

### 🙂 Expected behavior

Spreading an array inside an object where an array is expected should give a type error.

**Comment by val-o** [comment]
Duplicate of https://github.com/microsoft/TypeScript/issues/9726
